code/datums/datumvars.dm

```diff
@@ -748,6 +748,28 @@
 			src.give_disease(M)
 			href_list["datumrefresh"] = href_list["give_spell"]
 
+		else if(href_list["ninja"])
+			if(!check_rights(R_FUN))
+				return
+
+			var/mob/M = locate(href_list["ninja"])
+			if(!istype(M))
+				to_chat(usr, "This can only be used on instances of type /mob")
+				return
+
+			if(tgalert(usr, "Are you sure you want to make [M] into a ninja?", "Confirmation", "Yes", "No") == "No")
+				return
+
+			if(!M.mind)
+				M.mind_initialize()
+
+			var/datum/antagonist/ninja/hiyah = M.mind.has_antag_datum(/datum/antagonist/ninja)
+			if(!hiyah)
+				hiyah = add_ninja(M)
+			if(hiyah)

```

Comment:
`else`

Reply:
add_ninja returns a /datum/antagonist/ninja, or if it faiks, it doesn't, thia behavior is intended and an if else will break it

Reply:
right, i missed it, my bad.